## *********************************************************
##
## File autogenerated for the camera_laser_calibration package
## by the dynamic_reconfigure package.
## Please do not edit.
##
## ********************************************************/

from dynamic_reconfigure.encoding import extract_params

inf = float('inf')

config_description = {'upper': 'DEFAULT', 'lower': 'groups', 'srcline': 246, 'name': 'Default', 'parent': 0, 'srcfile': '/opt/ros/melodic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'cstate': 'true', 'parentname': 'Default', 'class': 'DEFAULT', 'field': 'default', 'state': True, 'parentclass': '', 'groups': [], 'parameters': [{'srcline': 274, 'description': 'How to use!', 'max': '', 'cconsttype': 'const char * const', 'ctype': 'std::string', 'srcfile': '/opt/ros/melodic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Info', 'edit_method': '', 'default': 'When you choose Save, it will save the laser_coor and image octomatically!Remember Modify the laser_coor according rviz', 'level': 0, 'min': '', 'type': 'str'}, {'srcline': 274, 'description': 'The min angle of laser', 'max': 140, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/melodic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'range_min', 'edit_method': '', 'default': -140, 'level': 0, 'min': -140, 'type': 'int'}, {'srcline': 274, 'description': 'The max angle of laser', 'max': 140, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/melodic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'range_max', 'edit_method': '', 'default': 140, 'level': 0, 'min': -140, 'type': 'int'}, {'srcline': 274, 'description': 'x y', 'max': '', 'cconsttype': 'const char * const', 'ctype': 'std::string', 'srcfile': '/opt/ros/melodic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'laser_coor', 'edit_method': '', 'default': '0.0, 0.0', 'level': 0, 'min': '', 'type': 'str'}, {'srcline': 274, 'description': 'Save this data?', 'max': True, 'cconsttype': 'const bool', 'ctype': 'bool', 'srcfile': '/opt/ros/melodic/lib/python2.7/dist-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'Save', 'edit_method': '', 'default': False, 'level': 0, 'min': False, 'type': 'bool'}], 'type': '', 'id': 0}

min = {}
max = {}
defaults = {}
level = {}
type = {}
all_level = 0

#def extract_params(config):
#    params = []
#    params.extend(config['parameters'])
#    for group in config['groups']:
#        params.extend(extract_params(group))
#    return params

for param in extract_params(config_description):
    min[param['name']] = param['min']
    max[param['name']] = param['max']
    defaults[param['name']] = param['default']
    level[param['name']] = param['level']
    type[param['name']] = param['type']
    all_level = all_level | param['level']

